app/models/customer.rb

```diff
@@ -1,4 +1,8 @@
 class Customer < ApplicationRecord
-  belongs_to :product
-  belongs_to :dog
+  has_many :dogs
+  has_many :customer_products
+  has_many :products, through: :customer_products
```

Comment:
could you explain this?

Reply:
- A customer can have 0-many dogs.
- A customer can have 0-many entries in the CustomerProduct table. A customer is linked with a Product through the CustomerProduct table.